lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -56,8 +86,12 @@ def include_meta(json)
         end
 
         def resource_object_for(options)
-          cache_check(serializer) do
-            serializer.attributes(options[:fields])
+          cached_serializer = CachedSerializer.new(serializer)
+
+          cached_attributes(cached_serializer) do
+            cached_serializer.cache_check(self) do
```

Comment:
Yeah, it the switch from `cache_check(serializer) do` to `cached_serializer = CachedSerializer.new(serializer); cached_serializer.cache_check(self) do` that isn't immediately obvious and I'm writing rather than digging into it :)